### ISSUE TITLE
Stratified sampling for dataframe splitting

### ISSUE THREAD

**Issue by codeheart09** [javascriptdata/scikit.js#191]
Hello everyone,

I'm working on a project that needs stratified sampling of the dataset so it can have a more balanced test set.
More on the subject: https://en.wikipedia.org/wiki/Stratified_sampling 

I implemented a solution using Danfo.js for that purpose and, if you think it is a good idea, I can open a PR with that as a splitting tool.
Its parallel in scikit learn: https://scikit-learn.org/stable/modules/generated/sklearn.model_selection.StratifiedShuffleSplit.html

If you think this would make sense for the project, just let me know. :)

**Comment by risenW** [comment]
@dcrescim, @codeheart09 is a new contributor and suggested stratified sampling feature.

**Comment by dcrescim** [comment]
Awesome @codeheart09 ! Yeah just throw up a PR, and we'd be happy to check it out and get it merged 😄

**Comment by codeheart09** [comment]
Hey! Working on it ;) Plan to open the PR soon!
